ocaml/xapi/xapi_globs.ml

```diff
@@ -1114,19 +1122,38 @@ let xapi_globs_spec =
   ; ("max_spans", Int max_spans)
   ; ("max_traces", Int max_traces)
   ; ("max_observer_file_size", Int max_observer_file_size)
+  ; ("test-open", Int test_open) (* for consistency with xenopsd *)
   ]
 
+let timeout ref str = ref := Xapi_stdext_unix.Unixext.Timeout.of_string str
+
+let span ref str =
+  ref := (Xapi_stdext_unix.Unixext.Timeout.(of_string str) :> Mtime.Span.t)
+
 let options_of_xapi_globs_spec =
   List.map
     (fun (name, ty) ->
       ( name
-      , (match ty with Float x -> Arg.Set_float x | Int x -> Arg.Set_int x)
+      , ( match ty with
+        | Float x ->
+            Arg.Set_float x
+        | Int x ->
+            Arg.Set_int x
+        | Timeout x ->
+            Arg.String (timeout x)
+        | Span x ->
+            Arg.String (span x)
```

Comment:
I'd prefer to keep using Arg.Float here, I don't like having the conversion being made deep in the stack call.

```suggestion
let span ref x =
  let t = Float.to_int (1000. *. x) in
  ref := Mtime.Span.(t * ms)

let options_of_xapi_globs_spec =
  List.map
    (fun (name, ty) ->
      ( name
      , (match ty with Float x -> Arg.Set_float x | Int x -> Arg.Set_int x)
      , ( match ty with
        | Float x ->
            Arg.Set_float x
        | Int x ->
            Arg.Set_int x
        | Timeout x ->
            Arg.String (timeout x)
        | Span x ->
            Arg.Float (span x)
```